### DISCUSSION TITLE
i3 will end?

### DISCUSSION THREAD

**Discussion by Fuhrerbaumanki** [i3/i3#5845] [Q&A] [answered]
Recently, there has been a noticeable trend of transition from x11 to wayland. Since i3 does not support wayland, it seems that the project will lose its relevance, tell me, does i3 have any chance of “survival”?

**Comment by hughdunc** [comment]
People will probably go yo Swaywm.

**Reply by okraits** [reply]
As long as some people will use X11, there will be some people who use i3. And I guess @stapelberg and the other maintainers will maintain i3 as long as it makes sense to them.

**Reply by stapelberg** [reply, marked as answer]
Yes, we’ll keep the project running as long as there are users. 

I would imagine that even as people switch from Xorg to Wayland, there will still be use-cases such as virtual desktop sessions via VNC or similar, where a light-weight X11 window manager will be useful. 

Then there also is the issue of Wayland and nVidia graphic cards. As long as that setup doesn’t work well (and it doesn’t, today), there will be many thousands of users who are stuck on Xorg, even if distributions have aggressive timelines for removal.

**Reply by Fuhrerbaumanki** [reply]
I agree with your arguments 100 percent, I recently tested hyprland on Nvidia myself and the result was sad to say the least. But wayland will one day overcome the problems with Nvidia and the fate of i3 will be under threat again, I wonder if the transition of i3 to wayland is possible in the future or is it technically impossible, for example, KDE has been working on these 2 protocols for a long time, is such a turn of events possible with i3?

**Reply by stapelberg** [reply]
I don’t see a point in transitioning i3 to Wayland, given that sway exists and provides a relatively smooth transition path.

It will probably be better for everyone if i3 stays on X11, and sway only works on Wayland.

**Reply by muihlinn** [reply]
I don't see any point in leaving X11 yet, given the way I use computers wayland has nothing to offer me, but it even breaks most of my configurations  which depends on having windows names, and that's not only i3, but some custom hardware I don't plan to leave behind.

**Comment by grafov** [comment]
I suppose X11 still remain an actual platform for years. Wayland looks like a right way for the future desktop but still lacks complete support from a lot of software. Especially proprietary software. I found problems with screen sharing between Wayland/X11 apps and virtual camera issues with v4l2loopback. Finally I've just switched back to X11 after 3 years on Wayland with Sway & River WMs.

**Reply by Saurabh254** [reply]
I agree. I encountered problems setting up the nvidia driver and  virtual camera on Wayland, so I switched back to X11.

**Comment by Thaodan** [comment]
From my point of view Sway or Wayland isn't 100% on par with I3 or other X11 window managers.

**Comment by boydthomson** [comment]
After nearly a decade of i3, I've gotten myself a new laptop and am considering Sway.  This may be a biased place to ask... but, should I stay the course, or start fresh with Sway?

**Reply by okraits** [reply]
I think this question relates not only to i3 but to all the programs you use - they need to work under Wayland without big issues.
Personally, I'm interested in Wayland and follow the developments but currently I don't feel like trying it out and solving potential issues.

**Comment by exaroth** [comment]
Personally I have no reason to migrate (yet), have to try Sway at some point though. Been using i3 for so long it's almost scary to install any other wm manager at this point ;)
